FlowCrypt/Functionality/WKDURLs/PubLookup.swift

```diff
@@ -24,35 +24,20 @@ class PubLookup: PubLookupType {
         self.attesterApi = attesterApi
     }
 
-    func lookup(with email: String) -> Promise<Contact> {
-        Promise<Contact> { resolve, _ in
-            let keyDetails = try awaitPromise(self.getKeyDetails(email))
-            // TODO: - we are blindly choosing .first public key, in the future we should return [Contact]
-            // then eg encrypt for all returned Contacts
-            // also stop throwing below - no point. Return
-            //  empty array then handle downstream
-            guard let keyDetail = keyDetails.first else {
-                throw ContactsError.keyMissing
-            }
-            resolve(Contact(email: email, keyDetail: keyDetail))
-        }
-    }
-
-    private func getKeyDetails(_ email: String) -> Promise<[KeyDetails]> {
-
-        Promise<[KeyDetails]> { [weak self] resolve, _ in
+    func lookup(with email: String) -> Promise<RecipientWithPubKeys> {
+        Promise<RecipientWithPubKeys> { [weak self] resolve, _ in
             guard let self = self else {
-                resolve([])
+                resolve(RecipientWithPubKeys(email: email, keyDetails: []))
                 return
             }
 
             let wkdResult = try awaitPromise(self.wkd.lookupEmail(email))
             if !wkdResult.isEmpty {
-                resolve(wkdResult)
+                resolve(RecipientWithPubKeys(email: email, keyDetails: wkdResult))
             }
```

Comment:
I suppose it's missing a return here after the resolve. I'm not sure if it makes any difference, but it makes me feel better :-)